src/Elmish.WPF/Merge.fs

```diff
@@ -15,22 +15,58 @@ type DuplicateIdException (sourceOrTarget: SourceOrTarget, index1: int, index2:
   member this.Index2 = index2
   member this.Id = id
 
+type internal CollectionTarget<'a> =
+  { GetLength: unit -> int
+    GetAt: int -> 'a
+    Append: 'a -> unit
+    InsertAt: int * 'a -> unit
+    ReplaceAt: int * 'a -> unit
+    RemoveAt: int -> unit
+    Move: int * int -> unit
+    Clear: unit -> unit
+    Enumerate: unit -> 'a seq
+    BoxedCollection: unit -> obj }
+
+module internal CollectionTarget =
+  let map (f: 'a -> 'b) (f': 'b -> 'a) (ct: CollectionTarget<'a>) : CollectionTarget<'b> =
+    { GetLength = ct.GetLength
+      GetAt = ct.GetAt >> f
+      Append = f' >> ct.Append
+      InsertAt = Pair.mapAll id f' >> ct.InsertAt
+      ReplaceAt = Pair.mapAll id f' >> ct.ReplaceAt
```

Comment:
Define `Pair.map2`.  Then this slightly simplifies to
```fsharp
Pair.map2 f' >> ct.FunctionName
```

Reply:
Updated. Also renamed `f` and `f'` to `fOut` and `fIn` respectively to more closely align with other naming conventions.